### ISSUE TITLE
string raw values

### ISSUE THREAD

**Issue by tcurdt** [filom/ASN1Decoder#16]
For IAP receipt validation one needs to calculate a digest. I have it working but I am a little bit sceptical about the `identifierBytes` bytes below.

```
let deviceBytes = try Device.macAddress()
let opaqueBytes = [UInt8](self.opaqueValue)
let identifierBytes = [UInt8(12), UInt8(self.bundleIdentifier.count)] + [UInt8](self.bundleIdentifier.utf8)
let sha1Bytes = [UInt8](self.sha1)

let digestBytes = deviceBytes + opaqueBytes + identifierBytes
```

I have no idea why the two bytes before the string are actually needed. My guess is that's the `UTF8String` tag and hence the raw bytes of the value are what goes into the digest.

While my current version works - there is no easy way to get to the raw value directly, is there?

**Comment by filom** [comment]
The 2 byte before date is needed in the asn.1 format to define the utf8string. 
In general after reading the tags I discard some of the original row content to reduce memory consumption. 
I curious about the `Device.macAddress()` is it for macOS or iOS too?

**Comment by tcurdt** [comment]
@filom just for macOS. On iOS you are supposed to use a provided device identifier.

Obviously would be great to have a way to also get to those raw bytes.
Maybe a specific mode that does not throw things away?

**Comment by filom** [comment]
To have the row bytes would be possible with serialization/write methods. It's in my future todo list.

**Comment by filom** [comment]
I bit late, but fixed with #23
